src/main/scala/fi/oph/koski/suoritusjako/SuoritusjakoService.scala

```diff
@@ -42,9 +40,17 @@ class SuoritusjakoService(suoritusjakoRepository: SuoritusjakoRepository, oppija
   }
 
   def putByKokonaisuudet(oppijaOid: String, kokonaisuudet: List[SuoritusjakoPayload])(implicit koskiSession: KoskiSpecificSession): Either[HttpStatus, Suoritusjako] = {
-    val suoritusjako = addSuoritusjako(oppijaOid, List(), kokonaisuudet)
-    AuditLog.log(KoskiAuditLogMessage(KANSALAINEN_SUORITUSJAKO_LISAYS_SUORITETUT_TUTKINNOT, koskiSession, Map(oppijaHenkiloOid -> oppijaOid)))
-    suoritusjako
+    // Tässä toSet, koska bugin vuoksi sama tyyppi saattaa esiintyä listalla useammin kuin kerran
+    kokonaisuudet.map(_.tyyppi).toSet.toList match {
+      case List("suoritetut-tutkinnot") =>
+        addSuoritusjako(oppijaOid, List(), kokonaisuudet)
+          .tap(_ => AuditLog.log(KoskiAuditLogMessage(KANSALAINEN_SUORITUSJAKO_LISAYS_SUORITETUT_TUTKINNOT, koskiSession, Map(oppijaHenkiloOid -> oppijaOid))))
+      case List("aktiiviset-ja-paattyneet-opinnot") =>
```

Comment:
Tässä mietin, että miksi käytetään listaa jos kuitenkin mätsätään että listassa on vain yksi tietty alkio.

Reply:
Ajatuksena on, että jatkossa voisi samalla suoritusjakolinkillä jakaa myös useamman kokonaisuuden kerralla samalle taholle, ja tietokannassa näitä on siksi lista. Tämä käy varmasti paremmin järkeen OAuth-tyyppisessä flow:ssa, jossa datan haluaja voi pyytää samantien useammat eri tiedot.

Nyt ei vaan toistaiseksi tueta kuin yhtä.